### ISSUE TITLE
Pixy2 demo code broken

### ISSUE THREAD

**Issue by VonSzarvas** [parallaxinc/solo#436]
Both solo and solocup

https://learn.parallax.com/sites/default/files/content/AB-Blockly/Projects/Pixy2/Pixy2TerminalDisplay.svg

This image shows the lack of new-line after the last comment, before the #include statement

![2020-05-21 07_07_32-BlocklyProp Solo](https://user-images.githubusercontent.com/11635798/82525992-0bb0e080-9b33-11ea-9705-a8eed07d92be.png)

solo shows blank code page and doesn't allow compile.
solocup allows compile with this error : Compile... Failed!

-------- compiler messages --------
Included libraries: simpletools
single.c:41:1: error: unknown type name 'fdserial'
single.c: In function 'main':
single.c:63:3: warning: implicit declaration of function 'fdserial_open' [-Wimplicit-function-declaration]
single.c:63:12: warning: assignment makes pointer from integer without a cast [enabled by default]
single.c: In function 'pixy_request_version':
single.c:92:3: warning: implicit declaration of function 'fdserial_txChar' [-Wimplicit-function-declaration]
single.c: In function 'pixy_get_response':
single.c:150:5: warning: implicit declaration of function 'fdserial_rxChar' [-Wimplicit-function-declaration]
single.c:155:18: error: 'list' undeclared (first use in this function)
single.c:155:18: note: each undeclared identifier is reported only once for each function it appears in
single.c: In function 'extract_version':
single.c:166:10: error: 'list' undeclared (first use in this function)
single.c:168:5: warning: passing argument 1 of 'dprint' from incompatible pointer type [enabled by default]
/opt/parallax/simple-libraries/Learn/Simple Libraries/TextDevices/libsimpletext/simpletext.h:430:5: note: expected 'struct text_t *' but argument is of type 'int *'
single.c: In function 'extract_resolution':
single.c:190:10: error: 'list' undeclared (first use in this function)
single.c:192:5: warning: passing argument 1 of 'dprint' from incompatible pointer type [enabled by default]
/opt/parallax/simple-libraries/Learn/Simple Libraries/TextDevices/libsimpletext/simpletext.h:430:5: note: expected 'struct text_t *' but argument is of type 'int *'
single.c: In function 'extract_block':
single.c:217:10: error: 'list' undeclared (first use in this function)
single.c: In function 'extract_word':
single.c:297:10: error: 'list' undeclared (first use in this function)
single.c: In function 'show_signatures':
single.c:331:21: error: 'list' undeclared (first use in this function)
single.c:41:1: error: unknown type name 'fdserial'
single.c: In function 'main':
single.c:63:3: warning: implicit declaration of function 'fdserial_open' [-Wimplicit-function-declaration]
single.c:63:12: warning: assignment makes pointer from integer without a cast [enabled by default]
single.c: In function 'pixy_request_version':
single.c:92:3: warning: implicit declaration of function 'fdserial_txChar' [-Wimplicit-function-declaration]
single.c: In function 'pixy_get_response':
single.c:150:5: warning: implicit declaration of function 'fdserial_rxChar' [-Wimplicit-function-declaration]
single.c:155:18: error: 'list' undeclared (first use in this function)
single.c:155:18: note: each undeclared identifier is reported only once for each function it appears in
single.c: In function 'extract_version':
single.c:166:10: error: 'list' undeclared (first use in this function)
single.c:168:5: warning: passing argument 1 of 'dprint' from incompatible pointer type [enabled by default]
/opt/parallax/simple-libraries/Learn/Simple Libraries/TextDevices/libsimpletext/simpletext.h:430:5: note: expected 'struct text_t *' but argument is of type 'int *'
single.c: In function 'extract_resolution':
single.c:190:10: error: 'list' undeclared (first use in this function)
single.c:192:5: warning: passing argument 1 of 'dprint' from incompatible pointer type [enabled by default]
/opt/parallax/simple-libraries/Learn/Simple Libraries/TextDevices/libsimpletext/simpletext.h:430:5: note: expected 'struct text_t *' but argument is of type 'int *'
single.c: In function 'extract_block':
single.c:217:10: error: 'list' undeclared (first use in this function)
single.c: In function 'extract_word':
single.c:297:10: error: 'list' undeclared (first use in this function)
single.c: In function 'show_signatures':
single.c:331:21: error: 'list' undeclared (first use in this function)

**Comment by zfi** [comment]
This issue is an artifact from the removal of the PrettyPrint package, which was designed to format C source code, not JavaScript. The C source code generator was relying on that module to correct formatting errors introduced in the generator, such as the one we have here.

**Comment by zfi** [comment]
The latest code in the feature-modules branch corrects the missing line wrap in the C source. However, it also appears that the project is missing an array init block for the default 'list' variable. My sense is that older version of Solo and BlockyProp defaulted this to an array of 10 elements because it was the default, or built-in, variable.

My recommendation is to update the project to include the missing array initializer block.

**Comment by zfi** [comment]
This is the updated project file that includes the missing array initializer for the "init" array.

[Pixy2TerminalDisplay-Rev-1.svg.zip](https://github.com/parallaxinc/solo/files/4727593/Pixy2TerminalDisplay-Rev-1.svg.zip)

**Comment by zfi** [comment]
I am seeing several projects that appear to be missing a "list" array init block. This is odd because the "list" array variable is a built-in default variable. If we assume that these projects used to work, I am left with the possibility that the app used to manufacture the initialization code for the "list" array.

**Comment by PropGit** [comment]
That's a reasonable theory.  The code I sent (different thread) that you indicated had this problem certainly worked in BlocklyProp and Solo for months with no problem (actually, I think it still works in Solo).

**Comment by zfi** [comment]
Some good news. The "list" init block was not a default in v1.4.x, so the project attached to this issue still breaks there. I'm looking to see if we customized the Blockly core array code since the last update to BlockyProp Server.

**Comment by zfi** [comment]
I am going to verify that the first issue, missing cr/lf in the C source code, has been corrected. I will create a new issue to explore the issue where the project loader is missing or dropping some array references and, somehow, replacing them with the default array variable, "list".
